From linux-kernel-owner+greg=40kroah.com-S932320AbWJEWJd@vger.kernel.org Thu Oct  5 15:14:10 2006
From: Eric Sesterhenn <snakebyte@gmx.de>
Subject: USB: fix dereference in drivers/usb/misc/adutux.c
To: Randy Dunlap <rdunlap@xenotime.net>
Cc: linux-kernel@vger.kernel.org, netwiz@crc.id.au
Date: Fri, 06 Oct 2006 00:09:29 +0200
Message-Id: <1160086169.26057.1.camel@alice>

in two of the error cases, dev is still NULL,
and we dereference it. Spotted by coverity (cid#1428, 1429)

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Cc: Randy Dunlap <rdunlap@xenotime.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/adutux.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/misc/adutux.c
+++ gregkh-2.6/drivers/usb/misc/adutux.c
@@ -370,7 +370,8 @@ static int adu_release(struct inode *ino
 	retval = adu_release_internal(dev);
 
 exit:
-	up(&dev->sem);
+	if (dev)
+		up(&dev->sem);
 	dbg(2," %s : leave, return value %d", __FUNCTION__, retval);
 	return retval;
 }
